src/animated.rs

```diff
@@ -0,0 +1,423 @@
+    /// Stop offsets and color values.
+    pub stops: super::ColorStops,
+}
+
+impl Gradient {
+    /// Returns true if the value contains no animated properties.
+    pub fn is_fixed(&self) -> bool {
+        self.start_point.is_fixed() && self.end_point.is_fixed() && self.stops.is_fixed()
+    }
+
+    /// Evaluates the animated value at the given frame.
+    pub fn evaluate(&self, frame: f64) -> peniko::Brush {
+        let start = self.start_point.evaluate(frame);
+        let end = self.end_point.evaluate(frame);
+        let stops = self.stops.evaluate(frame).into_owned();
+        if self.is_radial {
+            let radius = (end.to_vec2() - start.to_vec2()).hypot();
+            let mut grad = peniko::Gradient::new_radial(start, radius as f32);
+            grad.stops = stops;
+            grad.into()
+        } else {
+            let mut grad = peniko::Gradient::new_linear(start, end);
+            grad.stops = stops;
+            grad.into()
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ColorStops {
+    pub frames: Vec<Time>,
+    pub values: Vec<Vec<f64>>,
+    pub count: usize,
+}
+
+impl ColorStops {
+    pub fn evaluate(&self, frame: f64) -> fixed::ColorStops {
+        self.evaluate_inner(frame).unwrap_or_default()
+    }
+
+    fn evaluate_inner(&self, frame: f64) -> Option<fixed::ColorStops> {
+        let ([ix0, ix1], t, easing, hold) = Time::frames_and_weight(&self.frames, frame)?;
+
+        let v0 = self.values.get(ix0)?;
+        let v1 = self.values.get(ix1)?;
+
+        let mut stops: fixed::ColorStops = Default::default();
+        for i in 0..self.count {
+            let j = i * 5;
+            let offset = v0.get(j)?.tween(v1.get(j)?, t, &easing);
+            let t = if hold { 0.0 } else { t };
+            let r = v0.get(j + 1)?.tween(v1.get(j + 1)?, t, &easing);
+            let g = v0.get(j + 2)?.tween(v1.get(j + 2)?, t, &easing);
+            let b = v0.get(j + 3)?.tween(v1.get(j + 3)?, t, &easing);
+            let a = v0.get(j + 4)?.tween(v1.get(j + 4)?, t, &easing);
+            let stop = peniko::ColorStop::from((offset as f32, peniko::Color::rgba(r, g, b, a)));
+            stops.push(stop);
+        }
+        Some(stops)
+    }
+}
+
+/// Animated brush.
+#[derive(Clone, Debug)]
+pub enum Brush {
+    /// Solid color.
+    Solid(Value<Color>),
+    /// Gradient color.
+    Gradient(Gradient),
+}
+
+impl Brush {
+    /// Returns true if the value contains no animated properties.
+    pub fn is_fixed(&self) -> bool {
+        match self {
+            Self::Solid(value) => value.is_fixed(),
+            Self::Gradient(value) => value.is_fixed(),
+        }
+    }
+
+    /// Evaluates the animation at the specified time.
+    pub fn evaluate(&self, alpha: f64, frame: f64) -> fixed::Brush {
+        match self {
+            Self::Solid(value) => value.evaluate(frame).with_alpha_factor(alpha as f32).into(),
+            Self::Gradient(value) => value.evaluate(frame),
```

Comment:
Should this be using something like `brush_with_alpha` from fixed here?

Reply:
Likewise now that I think of it, this might not be correct.

`with_alpha_factor` will multiple the current alpha, not set the alpha. Those are two very different things.

Reply:
Will this need a further change to Peniko before it does its next release?

Reply:
It would be nice to see a `.set_alpha(alpha)` and rename `.with_alpha_factor(alpha)` to `.multiply(alpha)` or something

Reply:
@simbleau Please file a bug in Peniko so that we can resolve this before we do the new release.

Reply:
We're about to get to doing a Peniko release and this issue didn't get filed. Is it important enough to be concerned about it?